test/Controllers/ProductControllerTest.cs

```diff
@@ -0,0 +1,100 @@
+using Xunit;
+using Moq;
+using Services;
+using Entities;
+using Utils.Test;
+using System.Collections.Generic;
+namespace Controllers.Test;
+
+public class ProductControllerTest
+{
+    [Fact]
+    public void Add_OneProduct_VerifyAddCallOnce()
+    {
+        // Given
+        var product = ProductFixtures.BuildProduct();
+        var mockProductRepo = new Mock<IProductRepository>();
+        mockProductRepo.Setup(m => m.Add(It.IsAny<Product>())).Verifiable();
```

Comment:
Could you explain this?

Reply:
Here I'm creating a mock of the product repository interface, which I'll use some lines below to initialize the product controller. Also I'm setting up the behaviour of the `Add(Product)` method of my mocked `IProductRepository` and setting it as "verifiable" to test later if the controller called the add method of `mockProductRepo` using `mockProductRepo.Verify()`.